lib/src/authentication/sign_in/sign_in_view_model.dart

```diff
@@ -40,29 +50,54 @@ class SignInViewModel extends BaseViewModel {
     await router.push("/authentication/reset_password");
   }
 
-  Future<void> signIn() async {
+  Future<void> signIn(BuildContext context) async {
     if (formKey.currentState!.validate()) {
       // Change the sign in button into a loading.
       setBusyForObject(isSignInButtonEnabled, true);
 
       // Disable the form and buttons
       setBusy(true);
 
-      // TODO: Handle login
-      await router.pushReplacement(GlossaryView.routeName);
+      final bool isLoggedIn = await _userRepository.authenticate(
+        AuthenticationMethod.classic,
+        email: emailController.text,
+        password: passwordController.text,
+      );
+      if (isLoggedIn) {
+        emailController.clear();
+        passwordController.clear();
+        await router.pushReplacement(GlossaryView.routeName);
+      } else {
+        // ignore: use_build_context_synchronously
+        _toasterService.toast(context, l10n.sign_in_error);
```

Comment:
Don't forget to remove the loading if the login fails. Can you add a test to validate that please?

```suggestion
        _toasterService.toast(context, l10n.sign_in_error);
        setBusyForObject(isSignInButtonEnabled, false);
        setBusy(false);
```